Core/NemaGenerator.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *  @copyright Copyright 2021 The J-PET Monte Carlo Authors. All rights reserved.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may find a copy of the License in the LICENCE file.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  @file NemaGenerator.h
+ */
+
+#ifndef NEMAGENERATOR_H
+#define NEMAGENERATOR_H 1
+
+#include <G4PrimaryVertex.hh>
+#include <TVector3.h>
+#include <TGraph.h>
+#include <vector>
+#include <TF1.h>
+#include <map>
+
+struct NemaPoint
+{
+  NemaPoint();
+  
+  G4ThreeVector position;
+  G4double lifetime = 2;
+  bool is3GAllowed = false;
+  G4ThreeVector sizeOfPoint;
+  G4ThreeVector orientationOfPoint;
+  G4ThreeVector shapeOfPointInY;
+  TF1* elipseYNorm = nullptr;
+};
+
+class NemaGenerator
+{
+public:
+  NemaGenerator() {
```

Comment:
Does it follow the same seeding rules as generation of other random numbers in the simulations?
I recall at some point we added some control of whether the random sequences should be deterministic, different between simultaneous processes etc.
It would be good to use it here as well.